### ISSUE TITLE
Haplosaurus outputs nothing for some variants even with correct GT field

### ISSUE THREAD

**Issue by LukeGoodsell** [Ensembl/ensembl-vep#494]
Hi, I have several variants for which Haplosaurus outputs nothing and I cannot understand why. I have seen #315 but this is not the situation here, as I have formatted the GT field as required.

Example:

    $ cat example.vcf
    ##fileformat=VCFv4.2
    ##contig=<ID=chr14,length=107349540>
    ##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotypes">
    #CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	sample
    chr14	106511873	.	C	T	40	PASS	.	GT	0|1
    chr16	3817720	.	C	CT	40	PASS	.	GT	0|1
    
    haplo \
            --input_file example.vcf \
            --output_file example.json \
            --assembly GRCh37 \
            --refseq \
            --json \
            --database \
            --port 3337

I am using e!91.

**Comment by ens-lgil** [comment]
Dear @LukeGoodsell,

It looks like, in your 2 examples, that the variants don't overlap a NM/XM:
- 14: https://grch37.ensembl.org/Homo_sapiens/Location/View?db=core;r=14:106511870-106511880
- 16: https://grch37.ensembl.org/Homo_sapiens/Location/View?db=core;r=16:3817715-3817725

Because Haplosaurus only returns results on transcripts, in your examples Haplosaurus has nothing to report.

I just run this input, adding an extra variant on chromosome 14:

```
##fileformat=VCFv4.2
##contig=<ID=chr14,length=107349540>
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotypes">
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	sample
chr14	96152853	rs1360325170	T	C	40	PASS	.	GT	0|1
chr14	106511873	test_14	C	T	40	PASS	.	GT	0|1
chr16	3817720	test_16	C	CT	40	PASS	.	GT	0|1
```
and I got a result only for `rs1360325170` (using the default output format):
```
NM_004918.3	NM_004918.3:49T>C		NP_004909.1:17W>R			rs1360325170	sample:1
```
So basically Haplosaurus outputs nothing because it has nothing to report.

However I understand that the missing output file is confusing and we will try to add a warning message in case there is nothing reported by Haplosaurus.

Best regards,
Laurent

**Comment by LukeGoodsell** [comment]
Hi Laurent,

Thank you for your reply.

>  the variants don't overlap a NM/XM:

That is not what VEP reports for me:

    vep \
            --input_file "example.vcf" \
            --output_file "example.json" \
            --assembly "GRCh37" \
            --refseq \
            --json \
            --hgvs \
            --exclude_predicted \
            --use_transcript_ref \
            --check_existing \
            --database \
            --port 3337

    jq -rs '
            [.[] | {
                    chrom: .seq_region_name, pos: .start,
                    HGVSc: .transcript_consequences[].hgvsc,
                    HGVSp: .transcript_consequences[].hgvsp
            }]
                | (.[0] | to_entries | map(.key)), (.[] | [.[]]) | @csv' \
         example.json | csvlook

| chrom |         pos | HGVSc                                | HGVSp                                  |
| ----- | ----------- | ------------------------------------ | -------------------------------------- |
| chr14 | 106,511,873 | NC_000014.8:IGHV3-6:u_t_1.1:c.256G>A | NC_000014.8:IGHV3-6:u_t_1.1:p.Asp86Asn |
| chr16 |   3,817,721 | NM_001079846.1:c.3136dup             | NP_001073315.1:p.Ile1046AsnfsTer3      |
| chr16 |   3,817,721 | NM_001079846.1:c.3136dup             | NP_004371.2:p.Ile1084AsnfsTer3         |
| chr16 |   3,817,721 | NM_004380.2:c.3250dup                | NP_001073315.1:p.Ile1046AsnfsTer3      |
| chr16 |   3,817,721 | NM_004380.2:c.3250dup                | NP_004371.2:p.Ile1084AsnfsTer3         |

The chr14:106511873 C>T is in a pseudogene, but the chr16:3817720 C>CT is not.

**Comment by ens-lgil** [comment]
Dear @LukeGoodsell,

I see that VEP reports different informations.
However Haplosaurus works differently as it is focused on the resulting transcript/protein sequences.

Regarding the variant on `chr14` it overlaps `NC_000014.8:IGHV3-6` which is a pseudogene. Haplosaurus only provide result for protein coding transcripts, that's why nothing is returned for this variant.

About the other variant on `chr16`, it falls just outside the exons of `NM_001079846.1`,  `NM_004380.2`, `XM_005255124.1` and `XM_005255125.1`.
It is true that the variant is labelled as `splice_region_variant` but Haplosaurus is not able at the moment to predict the sequence of the transcript based on a splice region change.

We definitively need to improve the documentation about Haplosaurus.

I hope this makes sense.
Best regards,
Laurent

**Comment by LukeGoodsell** [comment]
Thanks again, @ens-lgil,

So would it be correct to say that Haplosaurus interprets chr16:3817720 C>CT as falling (just) outside exon of NM_001079846.1 and NM_004380.2, and so therefore has no protein consequence?

**Comment by ens-lgil** [comment]
Dear @LukeGoodsell,

Yes, you are correct.

Best regards,
Laurent

**Comment by LukeGoodsell** [comment]
Thanks, @ens-lgil.
